lib/matplotlib/dates.py

```diff
@@ -243,6 +243,30 @@ def _to_ordinalf(dt):
 _to_ordinalf_np_vectorized = np.vectorize(_to_ordinalf)
 
 
+def _dt64_to_ordinalf(d):
+    """
+    Convert `numpy.datetime64` or an ndarray of those types to Gregorian
+    date as UTC float.  Roundoff is via float64 precision.  Practically:
+    microseconds for dates between 290301 BC, 294241 AD, milliseconds for
+    larger dates (see `numpy.datetime64`).  Nanoseconds aren't possible
+    because we do times compared to ``0001-01-01T00:00:00`` (plus one day).
+    """
+
+    zerod = False
+    if isinstance(d, np.datetime64):
+        zerod = True
+        d = np.atleast_1d(d)
```

Comment:
It runs for me without converting a single item into array and back. This is needed for some older numpy (mine is `1.13.3`)?

Reply:
Hmmm, I didn't even try!  You're right..